compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -240,6 +278,51 @@ impl Ssa {
             if let Some(mut generated_acir) =
                 context.convert_ssa_function(&self, function, brillig)?
             {
+                // We want to be able to insert Brillig stdlib functions anywhere during the ACIR generation process (e.g. such as on the `GeneratedAcir`).
+                // As we do want a reference to the `SharedContext` on the generated ACIR itself,
+                // we instead store the opcode location at which a Brillig call to a std lib function occurred.
+                // We then defer resolving the function IDs of those Brillig functions to when we have generated Brillig
+                // for all normal Brillig calls.
+                for (opcode_location, brillig_stdlib_func) in
+                    &generated_acir.brillig_stdlib_func_locations
+                {
+                    let mut call_to_existing_bytecode = None;
+                    if let Some(generated_pointer) =
+                        shared_context.generated_brillig_stdlib_pointer(brillig_stdlib_func)
+                    {
+                        // We cannot insert in the Brillig calls to resolve map here as we are doing an immutable borrow
+                        // to the shared context to fetch the pre-existing generated pointer.
+                        call_to_existing_bytecode = Some((*opcode_location, *generated_pointer));
```

Comment:
```suggestion
                        call_to_existing_bytecode = Some((*opcode_location, generated_pointer));
```
Would you be able to insert the call now that generated_pointer is no longer a reference?
It looks like you could also have done:
```rs
let generated_pointer = *generated_pointer;
```
previously then set `add_call_to_resolve` after?

Reply:
yeah you are correct I believe I will update this

Reply:
This logic is inside of a `SharedContext` method now